### PR TITLE
Moving Source StatsReporter to its own package

### PR DESCRIPTION
**Pull request by nachocano**
- Similar to the the webhook statsReporter, we are moving the source statsReporter to its own package. 
- Minor refactor for better readability.

### CODE REVIEW

File: source/source_stats_reporter_test.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func unregister() {
 }
 
 func TestStatsReporter(t *testing.T) {
-	t.Skip("Fails in PROW but not locally, needs further investigation")
+	t.Skip("Fails in PROW but not locally, possibly due to the webhook stats_reporter test. Needs further investigation.")
```
Comment (vaikas-google) [inline source/source_stats_reporter_test.go]: Is there an issue tracking this work? If not, could you create one?

Comment (nachocano) [reply source/source_stats_reporter_test.go]: Done, https://github.com/knative/pkg/issues/665
Thanks!

**Pull request merged**